cmd/posix.go

```diff
@@ -98,13 +99,30 @@ func isDirEmpty(dirname string) bool {
 }
 
 // Initialize a new storage disk.
-func newPosix(diskPath string) (StorageAPI, error) {
-	if diskPath == "" {
+func newPosix(ep *url.URL) (StorageAPI, error) {
+	if ep == nil {
 		return nil, errInvalidArgument
 	}
-	suppliedDiskPath := diskPath
-	var err error
+	suppliedDiskPath := ep.String()
+	var diskPath string
+	// For windows ep.Path is usually empty
+	if runtime.GOOS == "windows" {
+		// For full URLs windows drive is part of URL path.
+		// Eg: http://ip:port/C:\mydrive
+		if ep.Scheme == "http" || ep.Scheme == "https" {
```

Comment:
You are right, i need to fix this behavior. The overarching problem is loosing command line arguments.